packages/eslint-plugin/tests/rules/non-nullable-type-assertion-style.test.ts

```diff
@@ -1,14 +1,11 @@
-import path from 'path';
```

Comment:
the changes here were extracted from #5916 as an easy test for this behaviour.
could add a regression test against this, but I didn't bother because ultimately it's such a rare case.